reconcile/aws_support_cases_sos.py

```diff
@@ -0,0 +1,57 @@
+import logging
+
+import utils.gql as gql
+
+from reconcile.queries import AWS_ACCOUNTS_QUERY
+from reconcile.queries import GITLAB_INSTANCES_QUERY
+from utils.aws_api import AWSApi
+from utils.gitlab_api import GitLabApi
+
+
+def get_deleted_keys(accounts):
+    return {account['name']: account['deleteKeys']
+            for account in accounts
+            if account['deleteKeys'] not in (None, [])}
+
+
+def get_keys_to_delete(aws_support_cases):
+    search_pattern = 'We have become aware that the AWS Access Key '
+    keys = []
+    for account, cases in aws_support_cases.items():
+        for case in cases:
+            comms = case['recentCommunications']['communications']
+            for comm in comms:
+                body = comm['body']
+                split = body.split(search_pattern, 1)
+                if len(split) == 2:  # sentence is found, get the key
+                    key = split[1].split(' ')[0]
+                    keys.append({'account': account, 'key': key})
```

Comment:
I prefer using tuples for this.

Reply:
i'm leaving room for future work.
in addition, this is how we do it in other integrations as well (using dictionaries). wdyt?

Reply:
true, it's just that in this case it's a nice fit, but no need to do this now, I agree